front-end/src/app/contacts/contact-detail/contact-detail.component.ts

```diff
@@ -194,11 +200,22 @@ export class ContactDetailComponent implements OnInit {
     this.formSubmitted = false;
   }
 
-  private isFormInvalid(): boolean {
-    const type: ContactTypes = this.form?.get('type')?.value;
-    return Contact.getFieldsByType(type).reduce(
-      (isInvalid: boolean, fieldName: string) => isInvalid || !!this.form?.get(fieldName)?.invalid,
-      false
-    );
+  /**
+   * Given the type of contact given, return the appropriate JSON schema doc
+   * @param {ContactTypes} type
+   * @returns {JsonSchema} schema
+   */
+  private getSchemaByType(type: ContactTypes): JsonSchema {
```

Comment:
@toddlees Per our chat, moving the schema lookup into the contact model makes a lot of sense at first glance but because of wanting to keep the schema loading in the contact module for lazy loading strategies, we'll revisit this if that bit of code needs to be duplicated down the road. Great talking that issue through to look at alternatives